src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    let nextState = { ...currentState };
+
+    switch (action.type) {
```

Comment:
You can destructure all the needed props of action (type, extraData, keysToRemove)

Reply:
like this:

for (const { type, extraData, keysToRemove } of actions) {}